kernel/reduction.mli

```diff
@@ -118,9 +118,12 @@ val betazeta_appvect : int -> constr -> constr array -> constr
 
 val dest_prod       : env -> types -> Context.Rel.t * types
 val dest_prod_assum : env -> types -> Context.Rel.t * types
+val dest_lam        : env -> types -> Context.Rel.t * types
```

Comment:
Just realizing that, but the implicit documentation that the argument and result are types is wrong. This function operates on terms, not types, and hence should be using `constr`. Same for the function below that was there before the patch.